src/v/cluster/members_backend.cc

```diff
@@ -942,9 +934,8 @@ ss::future<std::error_code> members_backend::reconcile() {
 
     // execute reallocations
     co_await ss::parallel_for_each(
-      meta.partition_reallocations,
-      [this](partition_reallocation& reallocation) {
-          return reallocate_replica_set(reallocation);
+      meta.partition_reallocations, [this](auto& pair) {
+          return reallocate_replica_set(pair.first, pair.second);
```

Comment:
As discussed at the standup, it is a bit hard to wrap your head around if the current state of a reallocation is up to date because this function (probably should be called `reconcile_reallocation_state` because it does much more than just reallocate :)) is called near the end of the reconciliation loop iteration. Not a blocker but would be great to make it easier to follow.